lib/react_client/react_interop.dart

```diff
@@ -320,8 +320,8 @@ external ReactClass createReactDartComponentClass(
 /// the JS and Dart component instances.
 @JS('_createReactDartComponentClass2')
 external ReactClass createReactDartComponentClass2(
-    ReactDartInteropStatics2 dartInteropStatics, ComponentStatics<Component2> componentStatics,
-    [JsComponentConfig2 jsConfig]);
+    ReactDartInteropStatics2 dartInteropStatics, ComponentStatics2<Component2> componentStatics,
```

Comment:
Shouldn't be any reason we need to provide a generic here since we're making a "v2" version, right?